largeobjects.go

```diff
@@ -108,19 +109,19 @@ type LargeObjectOpts struct {
 }
 
 type LargeObjectFile interface {
-	io.Writer
 	io.Seeker
-	io.Closer
+	Write(ctx context.Context, p []byte) (n int, err error)
+	Close(ctx context.Context) error
```

Comment:
@ncw I had to change the interface so all the functions accept the context but this way it won\t work with most of io utils :/

Not sure if this is ok with you? I was also thinking about leaving for the reader/writer functions function without context to implement the `io.Writer` and `io.Reader` but not sure if it's worth it. Also that would cause user to use everywhere normal functions and just for read/write the `WithContext` variants.

Also dropping the implementation of `io.Writer` caused the need to create the `ContextualWriter` as an adapter for the io.writer in buffered large file. See below.

Please let me know what you prefer or how would you like to deal with this.

Reply:
Hmm, that is unfortunate!

I think we need this to carry on implementing `io.Writer` and `io.Closer` otherwise that makes the objects dramatically less useful.

Perhaps we should implement a `WriteCtx` and `CloseCtx` method too?

Reply:
I agree about the usability. If a user does not care about the context it would make it way more complicated for him to use.

Ok, I'll add the and read functions to still implement the writer and reader interfaces. 
Would you prefer `WriteCtx` or `WriteWithContext`?

Reply:
The standard library prefers `WithContext` (2 votes) over `Ctx` (0 votes) so that is probably the most idiomatic - lets go with that!

Reply:
Great, adjusted as you suggested. Now again implements the `io.Writer` and has additional `WriteWithContext`.
Hopefully this way it's ok. I'm resolvimg this so PTAL